### DISCUSSION TITLE
Why the Message PMU B02 is missing in the message models?

### DISCUSSION THREAD

**Discussion by antoprd** [nHapiNET/nHapi#491] [Q&A] [answered]
I installed the model NHapi.Model.V27 version 3.0.3 then after an upgrade the message PMU B02 is not available anymore.

**Comment by milkshakeuk** [comment, marked as answer]
@antoprd 

I'm pretty sure a class model for `PMU_B02` has never existed. `nhapi` as well as `hapi` using mapping to map multiple message/event types to a single class, it does this using `EventMap.properties` files.

The below is taken from the  [hapi FAQ page](https://hapifhir.github.io/hapi-hl7v2/hapi-faq.html#missing_messages).
> HL7 defines that some message triggers reuse the same structure. So, for example, the ADT^A04 message has the exact same structure as an ADT^A01 message. Therefore, when an ADT^A04 message is parsed, or when you want to create one, you will actually use the ADT_A01 message class, but the "triggerEvent" property of MSH-9 will be set to A04.

[This](https://github.com/nHapiNET/nHapi/blob/30c14825cdde1ec4b682a46ce3abb0539b559a1a/src/NHapi.Model.V27/EventMapping/EventMap.properties#L52) is the EventMap for V27 which maps `PMU_B02` into the `PMU_B01` class.

I hope this helps.

**Reply by antoprd** [reply]
Thank you, that solved the problem.
